integration/auth/auth_test.go

```diff
@@ -53,12 +61,423 @@ func TestMain(m *testing.M) {
+		}
+	}
+}
+
+func testIterPage(t *testing.T) {
+	iter := client.Users(context.Background(), "")
+	pager := iterator.NewPager(iter, 2, "")
+	userCount := 0
+	pageCount := 0
+
+	for {
+		pageCount++
+		var users []*auth.ExportedUserRecord
+		nextPageToken, err := pager.NextPage(&users)
+		userCount += len(users)
+		if err != nil {
+			t.Errorf("paging error %v", err)
+		}
+		for _, u := range users {
+			// this iterates over users in a page
+			if u.UID == "something" {
+				// do something
+			}
+		}
+		if nextPageToken == "" {
+			break
+		}
+	}
+	if userCount < 5 || pageCount < 3 {
+		t.Errorf("expecting at least %d pages with at least %d users, got %d with %d ", 3, 5, pageCount, userCount)
+	}
+}
+
+func testDisableUser(t *testing.T) {
+	u, err := client.GetUser(context.Background(), testFixtures.sampleUserBlank.UID)
+	if err != nil {
+		t.Error(err)
+	}
+	if u.Disabled {
+		t.Errorf("expecting user not disabled")
+	}
+
+	_, err = client.UpdateUser(context.Background(), u.UID,
+		(&auth.UserToUpdate{}).Disabled(true))
+	if err != nil {
+		t.Error(err)
+	}
+	u, err = client.GetUser(context.Background(), testFixtures.sampleUserBlank.UID)
+	if !u.Disabled {
+		t.Errorf("expecting user disabled")
+	}
+	_, err = client.UpdateUser(context.Background(), u.UID,
+		(&auth.UserToUpdate{}).Disabled(false))
+	if err != nil {
+		t.Error(err)
+	}
+	u, err = client.GetUser(context.Background(), testFixtures.sampleUserBlank.UID)
+	if err != nil {
+		t.Error(err)
+	}
+	if u.Disabled {
+		t.Errorf("expecting user enabled")
+	}
+}
+
+func testUpdateUser(t *testing.T) {
+	u, err := client.GetUser(context.Background(), testFixtures.sampleUserBlank.UID)
+
+	if err != nil || u == nil {
+		t.Errorf("error getting user %s", err)
+	}
+	refU := &auth.UserRecord{
+		UserInfo: &auth.UserInfo{UID: testFixtures.sampleUserBlank.UID},
+		UserMetadata: &auth.UserMetadata{
+			CreationTimestamp: testFixtures.sampleUserBlank.UserMetadata.CreationTimestamp,
+		},
+	}
+	if !reflect.DeepEqual(u, refU) {
+		t.Errorf("\ngot %s, \nexpecting %s", toString(refU), toString(u))
+	}
+	utup := (&auth.UserToUpdate{}).
+		Disabled(false).
+		DisplayName("name").
+		PhoneNumber("+12345678901").
+		PhotoURL("http://photo.png").
+		Email("abc@ab.ab").
+		EmailVerified(true).
+		Password("wordpass").
+		CustomClaims(map[string]interface{}{"custom": "claims"})
+
+	_, err = client.UpdateUser(context.Background(), u.UID, utup)
+	if err != nil {
+		t.Error(err)
+	}
+
+	refU = &auth.UserRecord{
+		UserInfo: &auth.UserInfo{
+			UID:         testFixtures.sampleUserBlank.UID,
+			DisplayName: "name",
+			PhoneNumber: "+12345678901",
+			PhotoURL:    "http://photo.png",
+			Email:       "abc@ab.ab",
+		},
+		UserMetadata: &auth.UserMetadata{
+			CreationTimestamp: testFixtures.sampleUserBlank.UserMetadata.CreationTimestamp,
+		},
+		Disabled:      false,
+		EmailVerified: true,
+		CustomClaims:  map[string]interface{}{"custom": "claims"},
+	}
+	u, err = client.GetUser(context.Background(), u.UID)
```

Comment:
Don't make another GetUser call here. Instead check against the user returned by UpdateUser above.